pkg/reconciler/v1alpha1/revision/resolve.go

```diff
@@ -32,6 +34,40 @@ type digestResolver struct {
 	transport http.RoundTripper
 }
 
+const (
+	// Kubernetes CA certificate bundle is mounted into the pod here, see:
+	// https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/#trusting-tls-in-a-cluster
+	k8sCert = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+
+	// OpenShift cluster-created CA certificate bundle is mounted into the pod here, see:
+	// https://docs.openshift.com/online/dev_guide/secrets.html#service-serving-certificate-secrets
+	openShiftCert = "/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt"
```

Comment:
Could we resolve the speciality here by "just" loading all `crt` bundles in the `/var/run/secrets/kubernetes.io/serviceaccount` folder? My knowledge here is super limited with respect to what can land up in there, but it sounds to me its serviceaccount specific anyway so we might be able to trust all of this by default.

Reply:
I like that better than hardcoding this.  Perhaps run it by the GKE security folks tomorrow?

Reply:
So going forward there is work to allow things to be mounted in under users control (as part of a projected volume).

I don't know that it's safe to assume that all CRT files in this directory are safe to trust fo the use case in question.  Anyone using this as a consumer may be passing multiple in order to subdivide trust.  For instance, in the openshift case the service-ca means "I trust that DNS in the cluster is secure" (istio would be similar if they injected a CA this way).  However, if an infrastructure component is running in a knative serving mode, the infrastructure component may not trust `service-ca` and `ca` the same way (`ca` is supposed to be kube).

So wildcard seems risky, but we know that we want to allow people to be able to use these.

The reconciler may need to accept a list of host CAs to trust instead of just blindly trusting both kube and openshift and istio and *insert 3rd party here*.

Reply:
Turning it around - "who is the resolver supposed to trust?"  If the resolver is functioning as infrastructure, the infrastructure should be able to identify the list and let an admin control it.  If the resolver is functioning as the end user, the end user may need to define trust (although it's more likely the infrastructure gets to decide)

Reply:
Sounds like it would be functioning as infrastructure. Ideally we'd trust exactly the same set of registries as the underlying node so we can resolve anything that can be pulled. Anything less would be a bug, and anything more wouldn't be useful.

I'm leaning towards trusting only `ca` by default, since registry operators could add more certs via the `SSL_CERT_DIR` or `SSL_CERT_FILE` environment variables if `ca` is insufficient.